### PR TITLE
refactor(util): Refactor aur-check-update.py

### PR DESCRIPTION
**Pull request by 67au**
Refactor aur-check-update.py

### CODE REVIEW

File: utils/aur-check-update.py [hunk 1]
```diff
@@ -55,21 +59,19 @@ def from_str(cls, s: str) -> Version:
         return cls(pkgver, pkgrel, epoch)
 
     @classmethod
-    async def from_file(cls, file: str) -> Version:
-        async with aiofiles.open(file) as fin:
+    async def from_file(cls, file: str | Path) -> Version:
```
Comment (KunoiSayami) [inline utils/aur-check-update.py:62]: Is this standard syntax?
```suggestion
    async def from_file(cls, file: Union[str, Path]) -> Version:
```

File: utils/aur-check-update.py [hunk 2]
```diff
@@ -85,43 +87,64 @@ def __str__(self) -> str:
...
-                    await asyncio.create_subprocess_exec('git', 'pull')
-                logging.info('Upgrade %s from %s to %s', item, version, new_version)
-            else:
-                logging.info('Found update %s(%s) (local: %s)', item, new_version, version)
-            os.chdir('..')
+        if item.name.startswith('.') or not item.is_dir():
+            return
+        pkgbuild_file = item.joinpath('PKGBUILD')
+        if not pkgbuild_file.exists():
+            logging.warning(f'PKGBUILD not in {str(item)}, SKIP!')
```
Comment (KunoiSayami) [inline utils/aur-check-update.py:101]: Why use `str()` in  **formatted string literals**
```suggestion
            logging.warning(f'PKGBUILD not in {item}, SKIP!')
```

File: utils/aur-check-update.py [hunk 3]
```diff
@@ -85,43 +87,64 @@ def __str__(self) -> str:
...
+            new_version = Version.from_str(await resp.text())
+
+        if new_version == version:
+            return
+        elif new_version < version:
+            logging.warning(f'Warning: {version} is newer than AUR')
+            return
+        if item.joinpath('.git').exists():
+            if not dry_run:
+                await asyncio.create_subprocess_exec('git', '-C', f'{str(item)}', 'pull')
```
Comment (KunoiSayami) [inline utils/aur-check-update.py:117]:
```suggestion
                await asyncio.create_subprocess_exec('git', '-C', f'{item}', 'pull')
```

**Review by KunoiSayami — CHANGES_REQUESTED** [review]

**Review by KunoiSayami — APPROVED** [review]
LGTM

File: utils/aur-check-update.py (hunk 1 above)
Comment (67au) [reply utils/aur-check-update.py:62]: Please refer to this [PEP](https://www.python.org/dev/peps/pep-0585/#generics-erasure), and this synatx not working on py3.9 after `from __future__ import annotations` .

**Pull request merged**